src/lib/ChargeStation/index.ts

```diff
@@ -535,6 +535,11 @@ export class Session {
       this.carBatteryStateOfCharge +=
         (amountKwhToCharge / this.carBatteryKwh) * 100;
 
+      this.carBatteryStateOfCharge = Math.min(
+        this.carBatteryStateOfCharge,
+        100
+      );
```

Comment:
Another issue I noticed is the SoC could exceed 100%.. so also resolving that here.